### DISCUSSION TITLE
Add AI components to the lab

### DISCUSSION THREAD

**Discussion by sonwan2020** [Azure-Samples/java-microservices-aca-lab#17] [Ideas]
Enable a new component to embrace the power of AI

The AI context to this lab is TBD

Some reference resource

* https://github.com/Azure-Samples/spring-petclinic-langchain4j
* https://github.com/seanli1988/petclinic/tree/ai
* https://github.com/Azure-Samples/azure-openai-rag-workshop-java

Future investigation and discussions required

**Comment by frankliu20** [comment]
I would prefer this sample https://github.com/showpune/spring-petclinic-springai
with bellow reasons
- It leverages spring-ai lib, which is spring native
- It integrates with the pet-clinic microservice well
- Internally, it demonstrate popular AI usage patterns, including `RAG` and `Function calls` with very intuitive sample codes

**Reply by vermegi** [reply]
Is this the one we are working on together with the spring team? 
Asir mentioned something to me that there are joint efforts for this, which also leverage spring-ai. So if it's the same repo, I would say, yes, start from that one.

**Reply by vermegi** [reply]
@selvasingh can you comment whether this is the right code to start from? Or can you share the link from the spring team?

**Comment by vermegi** [comment]
When working this into the lab, would it be ok here as well to go for a step by step approach and not just give learners the finished product? Currently they are not really learning 'how' to do the AI bit, only thing they learn is the setup of the Azure services for it, but that's about it.

**Comment by vermegi** [comment]
The official sample to use for the AI lab: https://github.com/spring-petclinic/spring-petclinic-microservices/pull/281 
@frankliu20

**Comment by sonwan2020** [comment]
@vermegi we noticed the new PR about AI in petclinic.
- we already have our version of AI sample in the lab, see the latest [Lab 5 Integrate with Azure OpenAI](https://azure-samples.github.io/java-microservices-aca-lab/docs/05_lab_openai/05_openlab_openai_aca.html)
- technically, there are no big differences between these two versions of AI components

The screenshot of our version:
![image](https://github.com/user-attachments/assets/b6b9189e-6f6b-47bd-be87-e8098a74b30f)

I don't think we will replace our version with the new one.
